articles/password-checker-javascript/index.md

```diff
@@ -0,0 +1,230 @@
+
+### The Regular Expressions Class
+
+Regular expressions are patterns used to match character combinations in strings. In JavaScript, regular expressions are also objects.
+
+We can either use **simple patterns** such as `/vet/` for finding a direct much or **special characters** like `/?=/` for getting more than a direct match.
+
+### How to use Regex
+
+Regex, also denoted as `RegExp`, can be constructed in one of these two ways:
+
+- **Using the Regular Expression literal** where the pattern is enclosed between two slashes.
+
+```
+let check = /vet/
+```
+
+- **Using the constructor function** of the *RegExp()* class and passing the string to be checked as a parameter.
+
+```
+let check = new RegExp('vet')
+```
+
+We use the constructor function when the pattern keeps changing or if we get it from a different source, such as the user input. For this reason, we are going to use the constructor function since we are getting the password from the user input.
+
+#### Patterns we are going to use
+
+|Characters|Meaning|
+--|--|
+\d|Checks for a digit match e.g it returns 2 in "U2". |
+|\W|Checks for a special character e.g returns % in "2%".|
+|x{n,}|Checks for at least n terms from the preceding term x e.g O{2,} does not return anything in "BOY" but returns all Os in GOOOOOAL!.|
+|xIy |Matches either x or y in a string| eg YouIHim returns You in "You aren't lucky".
+|[^vet]|A negated set. Doesn't check for anything included in the range ie Does not check for *vet* in "veterinary"  |
+|[A-Za-z0-9]|Checks all alphanumeric characters|
+|[a-z]|Checks for lowercase letters|
+|[A-Z]|Checks for uppercase letters|
+|x(?=y)|Returns x if and only if it is followed by y|
+|**.**|Checks for any single character except line terminators |
+|x*|Checks for x 0 or more times|
+
+
+#### The JavaScript program
+
+We are going to discuss how to check the strength of a password that a user enters based on:
+
+- Is at least 8 characters long(`?=.{8,}`)
+
+- Has at least one uppercase letter(`?=.*[A-Z]`)
+
+- Has at least one lowercase letter(`?=.*[a-z]`)
+
+- Has at least one digit(`?=.*[0-9]`)
+
+- Has at least one special character(`[^A-Za-z0-9]`)
+
+
+We are going to have three levels:
+
+- **Strong**: The password has to meet all the requirements to be strong.
+
+```java
+(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{8,})
+```
+
+![Strong password](/engineering-education/password-checker-javascript/strong-password.png)
+
+- **Medium**: The password can only be medium either if it is at least 6 characters long and meets all the other requirements or has no digit but meets the rest of the requirements.
+
+```java
+((?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{6,}))|((?=.*[a-z])(?=.*[A-Z])(?=.*[^A-Za-z0-9])(?=.{8,}))
+```
```

Comment:
@Agusioma Can you provide step by step construction of these Regular Expressions.